pipeline/utils.py

```diff
@@ -51,9 +65,87 @@ def get_latest_provenance(process_name, session=None):
     return prov
 
 
-def normalize_header_key(key):
+def parse_dateobs(dateobs=None, output='datetime'):
+    """
+    Parse the dateobs, that can be a float, string, datetime or Time object.
+    The output is datetime by default, but can be any of the above types.
+    If the dateobs is None, the current time will be returned.
+    If float, will assume MJD.
+
+    Parameters
+    ----------
+    dateobs: float, str, datetime, Time or None
+        The dateobs to parse.
+    output: str
+        Choose one of the output formats:
+        'datetime', 'Time', 'float', 'str'.
+
+    Returns
+    -------
+    datetime, Time, float or str
+    """
+    if dateobs is None:
+        dateobs = Time.now()
+    elif isinstance(dateobs, (int, float)):
```

Comment:
We could also put in an if dateobs > 2400000 , and if so, interpret it as a JD, otherwise interpret it as an MJD.

(This codes in a Y2K-like bug that we'll hit in about 6000 years.)

Reply:
I like that!